spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/BinderHeaders.java

```diff
@@ -40,6 +40,12 @@
 	public static final String PARTITION_HEADER = "partition";
 
 	/**
+	 * Indicates the target partition of an outbound message. Overrides
+	 * any partition selected by the binder.
+	 */
+	public static final String PARTITION_OVERRIDE = "partitionOverride";
```

Comment:
Well, we always can do that and support both headers sets some time.
Like that was once in Spring Cloud Sleuth.